src/TableGateway/AbstractTableGateway.php

```diff
@@ -337,6 +337,15 @@ public function update($set, $where = null)
         if ($where !== null) {
             $update->where($where);
         }
+
+        if ($joins !== null) {
+            if (is_array($joins)) {
+                foreach ($joins as $join) {
+                    $type = isset($join['type']) ? $join['type'] : null;
```

Comment:
Should use `$update::JOIN_INNER` instead of `null` to get the default behavior. Will do that on merge.